src/react/components/Identity.tsx

```diff
@@ -0,0 +1,59 @@
+import { useAnchorWallet, useConnection } from "@solana/wallet-adapter-react";
+import React, { useEffect, useState } from "react";
+import { GatewayProvider } from "@civic/solana-gateway-react";
+import { getGatekeeperNetwork } from "../../client/api";
+import { Wallet } from "@project-serum/anchor";
+import { PublicKey } from "@solana/web3.js";
+import { config } from "../../config";
+import { Gateway } from "@components/Gateway";
+import { SolanaCluster } from "../../types/solana.types";
+
+export const Identity = () => {
+	const wallet = useAnchorWallet();
+	const connection = useConnection();
+
+	const [gatekeeperNetwork, setGatekeeperNetwork] = useState<PublicKey | undefined>(undefined);
+
+	useEffect(() => {
+		if (wallet?.publicKey && connection.connection) {
+			updateGatekeeperNetwork();
+		}
+	}, [wallet?.publicKey]);
+
+	const updateGatekeeperNetwork = async () => {
+		const gatekeeperNetwork = await getGatekeeperNetwork(connection.connection, wallet as Wallet);
+		setGatekeeperNetwork(gatekeeperNetwork);
+	};
+
+	const mapClusterNameToStage = (clusterName: SolanaCluster) => {
+		switch(clusterName) {
+			case SolanaCluster.LOCALNET: {
+				return "local";
+			}
+			case SolanaCluster.DEVNET: {
+				return "dev";
+			}
+			case SolanaCluster.MAINNET: {
+				return "prod";
+			}
+			default: {
+				break;
+			}
+		}
+	};
+
+	return (
+		<>
+			<GatewayProvider
```

Comment:
this will put al things in such as `mapClusterNameToStage` with the effect into the top level, we also only have
```
const wallet = useAnchorWallet();
const connection = useConnection();
```
this available after the `ConnectionProvider` tag in credixapp.tsx I guess but we already need it here, so not sure how to do it, I guess it's best to do it in call and see if that could work